x-pack/plugins/uptime/server/lib/synthetics_service/hydrate_saved_object.ts

```diff
@@ -19,38 +19,47 @@ export const hydrateSavedObjects = async ({
   server: UptimeServerSetup;
 }) => {
   try {
-    const missingUrlInfoIds: string[] = [];
+    const missingInfoIds: string[] = monitors
+      .filter((monitor) => {
+        const isBrowserMonitor = monitor.attributes.type === 'browser';
+        const monitorAttributes = monitor.attributes as MonitorFields;
+        const isMissingUrls = !monitorAttributes || !monitorAttributes.urls;
+        const isMissingPort = !monitorAttributes || !monitorAttributes['url.port'];
+        return isBrowserMonitor && (isMissingUrls || isMissingPort);
```

Comment:
I'd be happy to do that. It seems easy enough to include it in this PR, so I'll give it a try. I'll let you know in case there are unexpected complications which require this to be split into a separate issue, but I can't foresee any.